atlp-rwanda/Dominators-Phantom-BE
# Bg fixing develop

### What does this PR do?
- Fixes the error `Cannot set headers after they are sent to the client` while registering a new user
### Description of Task to be completed?

### How should this be manually tested?
- After cloning the repo run `npm i` to install all packages
- For Windows users refer an extra step for installing [`redis`](https://www.youtube.com/watch?v=188Fy-oCw4w&ab_channel=ProgrammingKnowledge2)
- use postman to test different endpoints present in [`index.js`](https://github.com/atlp-rwanda/Dominators-Phantom-BE/blob/develop/src/routes/index.js)
- Run `npm run migrate` to migrate all models in the local database
- Run `npm run seed` to populate the tables with seeders' values
- Run `npm run dev` to start the server manually
- use `http://localhost:3000/api/v1/login` to log in by using admin credentials present in the seeder of the user table. (Change the port **3000** to adhere to your chosen port)
- After login and adding Authorisation in headers, insert the URL with the port and the method e.g., `http://localhost:3000/api/v1/register` and `POST`
- The request body includes  two coordinates which consist of latitudes and longitudes of both origin and destination
``` 
{
    "firstName": "Jeie",
    "lastName": "Umusi",
    "email": "j.umuhire@alustudent.com",
    "role": "admin"
}
```
- The expected response should look like this
```
{
    "message": "user sucessfully registered",
    "data": {
        "id": "cb0c4894-3057-49e2-a774-b2c9a760668b",
        "firstName": "Jeie",
        "lastName": "Umusi",
        "email": "j.umuhire@alustudent.com",
        "role": "admin",
        "password": "$2a$12$.7KBHpjiwLGWuLMbJBG0ueG2iuyfRaRoe/q/zW53SJNhUdurLVwv.",
        "updatedAt": "2022-06-30T18:02:20.297Z",
        "createdAt": "2022-06-30T18:02:20.297Z"
    }
}
```

### Any background context you want to provide?

### Screenshots (if appropriate)

### Questions:
